pymeasure/experiment/results.py

```diff
@@ -376,20 +394,57 @@ def data(self):
                 pass  # All data is up to date
         return self._data
 
-    def reload(self):
-        """ Preforms a full reloading of the file data, neglecting
+    def reload(self, backupCount=None):
+        """ Performs a full reloading of the file data, neglecting
         any changes in the comments
+
+        :param backupCount: If a rotating file handler is used,
+                            how many backup files to load.
+                            If None, only load active file.
+                            If -1, load all. (Only loads consecutively numbered files.)
+                            (default: None)
         """
-        chunks = pd.read_csv(
-            self.data_filename,
-            comment=Results.COMMENT,
-            chunksize=Results.CHUNK_SIZE,
-            iterator=True
-        )
-        try:
-            self._data = pd.concat(chunks, ignore_index=True)
-        except Exception:
-            self._data = chunks.read()
+        # get data files
+        data_files = [ self.data_filename ]
+        if backupCount is None:
+            # needed to avoid comparison between None and integer
+            pass
+
+        elif backupCount > 0:
+            for i in range(backupCount):
+                filename = f'{self.data_filename}.{i+1}'
+                if os.path.isfile(filename):
+                    data_files.append(filename)
+
+        elif backupCount < 0:
+            i = 1
+            filename = f'{self.data_filename}.{i}'
+            while os.path.isfile(filename):
+                data_files.append(filename)
+                i += 1
+                file_name = f'{self.data_filename}.{i}'
+
+        data_files.reverse()  # oldest data has highest count name, but should be first in data
```

Comment:
We could overwrite the [`BaseRotatingHandler#doRollover`](https://github.com/python/cpython/blob/908fd691f96403a3c30d85c17dd74ed1f26a60fd/Lib/logging/handlers.py#L160) method, but I feel like it is better to work with how it is done rather than start mucking with that code.

Reply:
I see the advantage of using the handler as it is, but also for an up-counting filehandler; I'm not sure what the best approach is. A modification of the RotatingFileHandler doesn't seem to be very difficult though:

```
class forwardCountingRotatingFileHandler(RotatingFileHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.baseFilenamebase = self.baseFilename
        self.file_number = 0

    def doRollover(self):
        """
        Do a rollover, as described in __init__().
        """
        if self.stream:
            self.stream.close()
            self.stream = None

        self.file_number += 1
        self.baseFilename = self.rotation_filename("%s.%d" % (self.baseFilenamebase, self.file_number))

        if not self.delay:
            self.stream = self._open()
```

An additional advantage is that in this case the BackupCount (while still a valid keyword argument) doesn't have any impact anymore.